src/main/scala/fr/polytechnique/cmap/cnam/etl/transformers/interaction/NLevelInteractionTransformer.scala

```diff
@@ -0,0 +1,90 @@
+// License: BSD 3 clause
+
+package fr.polytechnique.cmap.cnam.etl.transformers.interaction
+
+import org.apache.spark.sql.Dataset
+import org.apache.spark.sql.functions
+import fr.polytechnique.cmap.cnam.etl.events.{Event, Exposure, Interaction}
+import fr.polytechnique.cmap.cnam.util.functions._
+
+case class NLevelInteractionTransformer(config: InteractionTransformerConfig) extends InteractionTransformer {
+  self =>
+
+  def joinTwoExposureNDataSet(right: Dataset[ExposureN], left: Dataset[ExposureN]): Dataset[ExposureN] = {
+    val sqlCtx = right.sqlContext
+    import sqlCtx.implicits._
+    right
+      .joinWith(
+        left,
+        left(Event.Columns.PatientID) === right(Event.Columns.PatientID) && !left("values").geq(right("values"))
+      )
+      .flatMap(e => e._1.intersect(e._2))
+      .repartition(functions.col("patientID"), functions.col("values"))
+      .cache()
+  }
+
+  private def wrapper(right: Dataset[ExposureN], left: Option[Dataset[ExposureN]]): Option[Dataset[ExposureN]] =
+    left.flatMap(l => Some(self.joinTwoExposureNDataSet(right, l)))
+
+  def elevateToExposureN(exposures: Dataset[Event[Exposure]], n: Int): List[Dataset[ExposureN]] = {
+    val sqlCtx = exposures.sqlContext
+    import sqlCtx.implicits._
+
+    val zero = Option(
+      sqlCtx
+        .createDataset(exposures.map(e => ExposureN(e.patientID, Set(e.value), Period(e.start, e.end.get))).rdd)
+        .repartition(functions.col("patientID"), functions.col("values")).cache()
+    )
+
+    List
+      .fill(n - 1)(
+        sqlCtx
+          .createDataset(exposures.map(e => ExposureN(e.patientID, Set(e.value), Period(e.start, e.end.get))).rdd)
+          .repartition(functions.col("patientID"), functions.col("values")).cache()
```

Comment:
Does this cache have the effect to improve the performance during the join operation ?